crates/sui-node/src/lib.rs

```diff
@@ -1115,6 +1119,44 @@ impl SuiNode {
         assert_eq!(next_epoch, new_epoch_store.epoch());
         new_epoch_store
     }
+
+    fn check_system_consistency(&self) {
+        if !self.config.enable_expensive_safety_checks && cfg!(not(debug_assertions)) {
+            return;
+        }
+        let total_storage_rebate: u64 = self
+            .state
+            .db()
+            .iter_live_object_set()
+            .map(|o| {
+                if let LiveObject::Normal(object) = o {
```

Comment:
Exhaustive pattern matching might save us some grief if we have multiple ways to represent a `LiveObject` w/ a storage rebate in the future.